airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/operation/AbstractStreamOperation.kt

```diff
@@ -278,25 +283,32 @@ abstract class AbstractStreamOperation<DestinationState : MinimumDestinationStat
         if (
             !isTruncateSync &&
                 syncSummary.recordsWritten == 0L &&
-                !initialRawTableStatus!!.hasUnprocessedRecords
+                !initialRawTableStatus.hasUnprocessedRecords
         ) {
             log.info {
                 "Skipping typing and deduping for stream ${streamConfig.id.originalNamespace}.${streamConfig.id.originalName} " +
                     "because it had no records during this sync and no unprocessed records from a previous sync."
             }
-        } else if (isTruncateSync && (!streamSuccessful || syncSummary.recordsWritten == 0L)) {
+        } else if (
+            isTruncateSync &&
+                (!streamSuccessful ||
+                    (syncSummary.recordsWritten == 0L &&
+                        !(initialRawTableStatus.rawTableExists &&
+                            initialRawTableStatus.hasUnprocessedRecords)))
```

Comment:
Alright, I think I understand: before if we crashed and started again but had no more records, we'd incorrectly interpret that as there being no intermediate state. This clarifies: we need both 0 new records AND the tmp table not to exist.

Reply:
yep. tried to capture that in the comment :/ but this is definitely an ugly conditional